### PR TITLE
PPCAnalyst: Don't discard registers across HLE'd functions

### PR DESCRIPTION
**Pull request by JosJuice**
Not sure if this was causing correctness issues – it depends on whether the HLE code was actually reading the discarded registers – but it was at least causing annoying assert messages in one piece of homebrew.

### CODE REVIEW

File: Source/Core/Core/HLE/HLE.h [hunk 1]
```diff
@@ -39,6 +39,14 @@ struct Hook
   HookFlag flags;
 };
 
+struct TryReplaceFunctionResult
+{
+  HookType type = HookType::None;
+  u32 hook_index = 0;
+
+  operator bool() const { return type != HookType::None; }
```
Comment (lioncash) [inline Source/Core/Core/HLE/HLE.h:47]:
```suggestion
  explicit operator bool() const { return type != HookType::None; }
```

**Comment by krnlyng** [comment]
Could this explain the issues i saw in https://github.com/dolphin-emu/dolphin/pull/10063 ?

**Comment by JosJuice** [comment]
Possibly. I'm still not quite sure what was going on in that PR, but the problem description of "The handler might use registers which were marked as discarded" matches.

**Review by lioncash — APPROVED** [review]

**Review by Pokechu22 — COMMENTED** [review]
Can confirm that this fixes the precompiled `aestest.elf` from devkitPro/libogc#143 (compiling it myself didn't cause the issue, either before or after; I'm not sure why.)

**Pull request merged**